purchase_order_product_recommendation_secondary_unit/wizards/purchase_order_recommendation.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2019 David Vidal <david.vidal@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+from odoo.tools.float_utils import float_compare, float_round
+from math import ceil
+
+
+class PurchaseOrderRecommendation(models.TransientModel):
+    _inherit = 'purchase.order.recommendation'
+
+    @api.model
+    def _prepare_wizard_line(self, vals, order_line=False):
+        res = super()._prepare_wizard_line(vals, order_line)
+        secondary_uom_id = (
+            order_line and order_line.secondary_uom_id or
+            vals.get('product_id') and
+            vals['product_id'].purchase_secondary_uom_id)
+        secondary_uom_qty = False
+        if not order_line and secondary_uom_id:
+            secondary_uom_qty = ceil(
+                res.get('units_included', 0) / secondary_uom_id.factor)
+            res['units_included'] = secondary_uom_qty * secondary_uom_id.factor
+        res.update({
+            'secondary_uom_id': secondary_uom_id and secondary_uom_id.id,
+            'secondary_uom_qty': (order_line and order_line.secondary_uom_qty
+                                  or secondary_uom_qty),
```

Comment:
Here it is also clearer to use `order_line.secondary_uom_qty if order_line else secondary_uom_qty` IMHO.